experimental/builder/include/ck_tile/builder/reflect/instance_traits_util.hpp

```diff
@@ -60,40 +60,25 @@ consteval std::string_view type_name()
 template <typename T>
 constexpr std::string_view layout_name()
 {
-    if constexpr(requires {
+    if constexpr(std::is_base_of_v<ck_tile::tensor_layout::BaseTensorLayout, T> && requires {
                      { T::name } -> std::convertible_to<std::string_view>;
                  })
         return T::name;
     else
-        static_assert(false, "layout type is missing name attribute");
+        static_assert(false,
+                      "Layout type must derive from BaseTensorLayout and have name attribute");
```

Comment:
I realized yesterday that this probably should have been

```suggestion
        static_assert(
            std::is_base_of_v<ck_tile::tensor_layout::BaseTensorLayout, T> && requires {
                { T::name } -> std::convertible_to<std::string_view>;
             },
             "Layout type must derive from BaseTensorLayout and have name attribute");
        return T::name;
```
I don't really think that its a problem though, but maybe cleaner or something. Maybe we should just update this next time that we're touching these files.

Reply:
Good suggestion. I think worth to add right now.